connectors/sources/sharepoint_online.py

```diff
@@ -1732,13 +1768,32 @@ def _get_id(permissions, identity):
 
             user_id = _get_id(granted_to_v2, "user")
             group_id = _get_id(granted_to_v2, "group")
+            site_group_id = _get_id(granted_to_v2, "siteGroup")
+            site_user_email = _get_email(granted_to_v2, "siteUser")
 
             if user_id:
                 access_control.append(_prefix_user_id(user_id))
 
             if group_id:
                 access_control.append(_prefix_group(group_id))
 
+            if site_user_email:
+                access_control.append(_prefix_email(site_user_email))
+
+            if site_group_id:
+                async for site_group_user in self.client.site_groups_users(
+                    site_web_url, site_group_id
+                ):
```

Comment:
IIUC, this will re-fetch the memberships of a given site group for every single drive item that is shared with that given site group.

This could be pretty inefficient and use up a lot of API quota. Would it be valuable to first check an in-memory cache to see if the members have already been looked up for this group, and if so, grab them from the cache, and if not, fetch them and add them to the cache?

Reply:
That's a great idea to save the API quota. One potential issue (apart from mem footprint) is the case connector keeps running for a long time. Whenever `get_docs` or `get_docs_incrementally` we need to invalidate the cache to fetch up-to-date group members

Reply:
You are right! For some reason I thought that the data source instance would be kept in memory inbetween the syncs. In that case we don't need to purge the cache. I will amend the PR